controllers/controlplane_controller.go

```diff
@@ -128,7 +128,9 @@ func (r *ControlPlaneReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	// Create or update the controllerVM CR object
 	ospControllerVM := &ospdirectorv1beta1.ControllerVM{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-controllervm", instance.Name),
+			// FIXME: currently we use the name of the resource as the basename
+			Name: "controller",
```

Comment:
@stuggi @abays in the BaremetalSet version of this patch (we merged last week) I had used the name of the resource as the basis of the hostname. So for baremetalSets 'compute', 'compute1', etc. For controllers we are creating just a single ctlplane right? And that creates the controlplane_vm's... so is it okay to just hard code this? Reason I ask is the alternate would be to use the name of the ControlPlane resource but we name it 'overcloud' so I'm not sure it fits...

Reply:
In my patch that I work on atm, I changed the baremetalset hostname to use the role name instead of the CR name,
which I think more alligns with the tripleo side. I planned to do the same for the controller, add a role param to the
CRD and use it for the hostname and set the HostnameFormat for the role. With this you could also use customized roles.

Reply:
@stuggi How were you planning on making hostname unique then? I had considered it, but as the 'Name' is guaranteed to be unique it seemed to fit instead.

Reply:
Looks like kubebuilder has validations that support unique. I've not tried them yet however...

Reply:
@dprince same like we have atm, rolename + index, just replacing instance.Name with the role name.